projects/packages/stats/src/class-wpcom-stats.php

```diff
@@ -437,11 +437,21 @@ protected function fetch_post_stats( $args, $post_id ) {
 		if ( $stats_cache ) {
 			$data = reset( $stats_cache );
 
-			if ( is_wp_error( $data ) ) {
+			if (
+				! is_array( $data )
+				|| empty( $data )
+				|| is_wp_error( $data )
```

Comment:
I wasn't sure, but it was the only check that was there before my change so I figured it may cover a use-case I haven't thought about.

I think I'll leave it here for now. Better be safe than sorry. :)